_overviews/scala3-contribution/arch-context.md

```diff
@@ -0,0 +1,43 @@
+---
+title: Contexts
+type: section
+description: This page describes symbols in the Scala 3 compiler.
+num: 18
+previous-page: arch-symbols
+next-page:
+---
+
+> (The following is work in progress), adapted from dotty.epfl.ch
+
+The `Context` contains the state of the compiler, for example
+
+  * `settings`
+  * `freshNames` (`FreshNameCreator`)
+  * `period` (run and phase id)
+  * `compilationUnit`
+  * `phase`
+  * `tree` (current tree)
+  * `typer` (current typer)
+  * `mode` (type checking mode)
+  * `typerState` (for example undetermined type variables)
+  * ...
+
+### Contexts in the typer
+The type checker passes contexts through all methods and adapts fields where
+necessary, e.g.
+
+```scala
+case tree: untpd.Block => typedBlock(desugar.block(tree), pt)(ctx.fresh.withNewScope)
+```
+
+A number of fields in the context are typer-specific (`mode`, `typerState`).
```

Comment:
What does it mean to be "typer-specific"?

Reply:
I'm not sure, this page was just copied from dotty website